lib/common/api/deprecate.js

```diff
@@ -69,23 +69,28 @@ const deprecate = {
     })
   },
 
-  promisify: (promise, cb) => {
-    const oldName = `function with callbacks`
-    const newName = `function with Promises`
-    const warn = warnOnce(oldName, newName)
+  promisify: (fn, numArgs) => {
+    return function (...args) {
+      const promise = fn.apply(this, args.splice(0, numArgs))
+      const cb = args[numArgs]
 
-    if (typeof cb !== 'function') return promise
-    if (process.enablePromiseAPIs) warn()
-    return promise
-      .then(res => {
-        process.nextTick(() => {
-          cb(null, res)
-        })
-      }, err => {
-        process.nextTick(() => {
-          cb(err)
+      const oldName = `function with callbacks`
+      const newName = `function with Promises`
```

Comment:
```js
`${fn.name} with callbacks`
`${fn.name} with Promises`
```

maybe?

Reply:
@ckerr `fn.name` isn't set for anonymous function, should defs default to `function` if we do want to use `fn.name`

Reply:
☝️ we don't have access to `fn.name`, which is why i did it in the above manner :)

Reply:
We appear to have it sometimes, e.g. we have it in the spec tests.

I've changed it to `fn.name || 'function'`